test/lib/auth.js

```diff
@@ -64,7 +64,192 @@ describe('Auth lib', function () {
         };
     };
 
-    describe('Update party API', function () {
+    describe('Set req.user object', function() {
+	var auth = getAuthLib(strategy, {}, null, {});
+	var reqInd = {
+	    id: 'test_user',
+	    appId: config.oauth2.clientID,
+	    user: {
+		id: 'rick',
+		nickName: 'theMagician',
+		accessToken: 'accesstoken',
+		refreshToken: 'refreshToken',
+		organizations: [{
+		    id: '123456789',
+		    name: 'patty',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '987654321',
+		    name: 'MntyPythn',
+		    roles: [{
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '111555999',
+		    name: 'AmaneceQueNoEsPoco',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '1',
+			name: 'purchuaser'
+		    }]
+		}]
+	    }
+        };
+
+	var reqOrg = {
+	    id: 'test_user',
+	    appId: config.oauth2.clientID,
+	    user: {
+		id: 'rick',
+		nickName: 'theMagician',
+		accessToken: 'accessToken',
+		refreshToken: 'refreshToken',
+		organizations: [{
+		    id: '123456789',
+		    name: 'patty',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '987654321',
+		    name: 'MntyPythn',
+		    roles: [{
+			id: '6',
+			name: 'provider'
+		    }]
+		}, {
+		    id: '111555999',
+		    name: 'AmaneceQueNoEsPoco',
+		    roles: [{
+			id: '3',
+			name: 'Seller'
+		    }, {
+			id: '1',
+			name: 'purchuaser'
+		    }]
+		}]
+	    },
+	    headers: {
+		'x-organization': '123456789'
+	    }
+        };
```

Comment:
The only difference between both requests is the headers stuff, maybe you can share the id, the token and the user object